src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -0,0 +1,562 @@
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="values" /> is <c>null</c>.</exception>
+        static virtual TSelf Create(T[] values) => TSelf.Create(values.AsSpan());
+
+        /// <summary>Creates a new vector from a given array.</summary>
+        /// <param name="values">The array from which the vector is created.</param>
+        /// <param name="index">The index in <paramref name="values" /> at which to being reading elements.</param>
+        /// <returns>A new vector with its elements set to the first <see cref="Count" /> elements from <paramref name="values" />.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">The length of <paramref name="values" />, starting from <paramref name="index" />, is less than <see cref="Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="values" /> is <c>null</c>.</exception>
+        static virtual TSelf Create(T[] values, int index) => TSelf.Create(values.AsSpan(index));
+
+        /// <summary>Creates a new vector from a given readonly span.</summary>
+        /// <param name="values">The readonly span from which the vector is created.</param>
+        /// <returns>A new vector with its elements set to the first <see cref="Count" /> elements from <paramref name="values" />.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">The length of <paramref name="values" /> is less than <see cref="Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static virtual TSelf Create(ReadOnlySpan<T> values)
+        {
+            if (values.Length < TSelf.Count)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.values);
+            }
+            return TSelf.LoadUnsafe(ref MemoryMarshal.GetReference(values));
+        }
+
+        /// <summary>Creates a new vector with the first element initialized to the specified value and the remaining elements initialized to zero.</summary>
+        /// <param name="value">The value that element 0 will be initialized to.</param>
+        /// <returns>A new vector with the first element initialized to <paramref name="value" /> and the remaining elements initialized to zero.</returns>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        static virtual TSelf CreateScalar(T value)
+        {
+            TSelf result = TSelf.Zero;
```

Comment:
Why is a local needed rather than e.g.
```C#
static virtual TSelf CreateScalar(T value) => TSelf.WithElement(TSelf.Zero, 0, value);
```
?

Reply:
It's not, it's just consistent with how existing code was already written/handled.

Reply:
Fixed.